### PR TITLE
Add libtoolbox

### PR DESCRIPTION
**Pull request by MementoRC**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MementoRC** [comment]
@conda-forge/help-c-cpp Hello, ready for review

File: recipes/libtoolbox/build.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -ex
+
+build_dir="${SRC_DIR}"/build-release
+test_release_dir="${SRC_DIR}"/test-release
+
+mkdir -p "${build_dir}"
+cd "${SRC_DIR}"/build-release
```
Comment (xhochy) [inline recipes/libtoolbox/build.sh:9]:
```suggestion
pushd "${SRC_DIR}"/build-release
```

File: recipes/libtoolbox/build.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  cmake -S "${SRC_DIR}" -B . \
+  -D CMAKE_BUILD_TYPE=Release \
+  -D CMAKE_INSTALL_PREFIX="${PREFIX}" \
+  -D toolbox_BUILD_SHARED_LIBS=ON \
+  -D toolbox_BUILD_TESTS=ON \
+  -G Ninja
+  cmake --build . -- -j"${CPU_COUNT}"
+  cmake --build . --target toolbox-test
+  cmake --install .
+cd "${SRC_DIR}"
```
Comment (xhochy) [inline recipes/libtoolbox/build.sh:19]:
```suggestion
popd
```

**Review by xhochy — COMMENTED** [review]
Small stlye nit, otherwise this can go in

**Comment by MementoRC** [comment]
@xhochy I seem to have encounter a ssituation where I need to intercept an `execve()` call made by `zig` when cross-compiling. I can do so by patching the official `qemu` git and when I replace the CROSSCOMPILING_EMULATOR with it, I get passed the :
```
error(link): unable to spawn /home/conda/feedstock_root/build_artifacts/zig_1721234700329/work/build-release/zig2: InvalidExe
error: UnableToSpawnSelf
```
caused by the `execve()` call. My 2 questions are:
   - would you approve of such a `subterfuge` (although zig is already a conda-forge package, I just try to add aarch64)?
   - would it be acceptable to create a conda-forge package the provides thie `qemu_execve`?

**Comment by xhochy** [comment]
Are you sure you commented on the right PR?

**Comment by MementoRC** [comment]
I am sure I did not ;P
But I am still struggling with the build and I have not made a PR yet

**Comment by xhochy** [comment]
Where does this error happen then? I do not see `zig` mentioned anywhere in the recipe.

**Comment by MementoRC** [comment]
Apologies, I meant I was asking a completely unrelated question.
The issue is in the zig-feedstock that I have been working on for several
months.

**Comment by xhochy** [comment]
Ah, then please mention me there! This goes into a separate queue on my local folders ;)

Is this PR ready for another review then?

**Comment by MementoRC** [comment]
Yes, it is ready. Thx

**Comment by MementoRC** [comment]
@xhochy I think it is good to go. I used it in the other recipe and realized that as-is, `find_package()` could not find it based upon the cmake files, so I revisited and added a duplicate location

**Review by xhochy — APPROVED** [review]

**Pull request merged**